### PR TITLE
Cddl conformance tests

### PR DESCRIPTION
**Pull request by redxaxder**
The tests that currently don't pass are commented out.

### CODE REVIEW

File: shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/CDDL.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import qualified Data.ByteString.Lazy as BSL
+
+type A = AllegraEra C_Crypto
+type M = MaryEra C_Crypto
+
+cddlTests :: Int -> TestTree
+cddlTests n = withResource combinedCDDL (const (pure ())) $ \cddl ->
+  testGroup "CDDL roundtrip tests" $
+    [ cddlTest @(Core.Value A) n "coin"
+    --, cddlTest @(Core.Value M) n "value"
```
Comment (uroboros) [inline shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/CDDL.hs:31]: I think make a TODO note here

**Review by uroboros — APPROVED** [review]
looks good

**Review by JaredCorduan — APPROVED** [review]
LGTM!

File: shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/CDDL.hs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import Cardano.Ledger.Allegra (AllegraEra)
+
+import Test.Tasty (TestTree, withResource, testGroup)
+import qualified Data.ByteString.Lazy as BSL
+
+type A = AllegraEra C_Crypto
+type M = MaryEra C_Crypto
+
+cddlTests :: Int -> TestTree
+cddlTests n = withResource combinedCDDL (const (pure ())) $ \cddl ->
```
Comment (nc6) [inline shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/CDDL.hs:28]: Might be nice to label what this argument is.

**Review by nc6 — APPROVED** [review]

**Pull request merged**